ssl/ssl_lib.c

```diff
@@ -2908,6 +2908,9 @@ long ossl_ctrl_internal(SSL *s, int cmd, long larg, void *parg, int no_quic)
     long l;
     SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(s);
 
+    if (sc == NULL)
+        return 0;
+
```

Comment:
I corrected the code so that pointer SSL is checked first. If it is equal to NULL, then returned 0, since pointer SSL_CONNECTION will also be NULL.
Next comes checking SSL_CONNECTION  for NULL only by ssl type. Pointer SSL_CONNECTION needs to be checked, since it will be dereferenced next.

Reply:
Please revert this to the original fix. In other cases we do not raise error for these failures and IMO there is no point doing that here anyway as these errors can happen only with severe application bugs.

Reply:
Reverted to the original fix.